### PR TITLE
Rsx: Remove vertex_draw_count (WIP)

### PR DESCRIPTION
**Pull request by vlj**
This PR makes indexing code simpler. It will remove vertex_draw_count member since it's rather confusing (sometimes it's used as index count, sometimes as vertex count, sometimes as stored vertex count...)

For now the PR only changes VK code. I'd like some testing for regression. If there is none I will replicate the changes on GL backend too.

### CODE REVIEW

File: rpcs3/Emu/RSX/VK/VKVertexBuffers.cpp [hunk 1]
```diff
@@ -204,33 +204,11 @@ namespace vk
...
-	}
 }
 
-std::tuple<VkPrimitiveTopology, bool, u32, VkDeviceSize, VkIndexType>
-VKGSRender::upload_vertex_data()
+namespace
 {
-	//initialize vertex attributes
-	const std::string reg_table[] =
+	constexpr std::array<const char*, 16> reg_table =
```
Comment (Nekotekina) [inline rpcs3/Emu/RSX/VK/VKVertexBuffers.cpp]: Just make is `static constexpr` and leave it inside the function

**Comment by vlj** [comment]
Renamed reg_table to s_reg_table and made static, rebased.

**Comment by Nekotekina** [comment]
Why another dependency?

**Comment by vlj** [comment]
Optional is a header only dependency which is part of the C++17 spec draft so it's not that intrusive.
It's also more self explanatory that a pointer imho.

**Comment by Nekotekina** [comment]
I mean, it doesn't look like something big, so. Could either be just one file with the trash cleaned up. Or even integrated somewhere.
Typing `std::experimental::` also looks terrible.

**Comment by vlj** [comment]
Moved optional.hpp to utilities instead of using a submodule and removed experimental namespace.

**Comment by Nekotekina** [comment]
Ok, thanks.

**Comment by raven02** [comment]
Compared with master , found 2 games regressed which is black screen at bootup with FPS running.

1. TNT racers
2. House of Dead 3

**Comment by vlj** [comment]
Does GL backend works with TNT racers and House of Dead 3 with this changes ?

**Comment by raven02** [comment]
Just tested and compared with master , GL backend also not working on these 2 games now.

**Comment by vlj** [comment]
I fixed TNTracers for both Vulkan and GL. Does House of Dead 3 works now ?

**Comment by SakataGintokiYT** [comment]
Fixed black screen in this game  / [ Vulkan backend ]
-The Awakened Fate Ultimatum

**Comment by raven02** [comment]
@vlj  House of Dead 3 also works okay now in GL and Vulkan

**Comment by danilaml** [comment]
Btw, `std::optional` should be included in the next update of MSVC, so we can remove this dependency soon.

**Pull request merged**